### PR TITLE
Adding preferredFrameRate to MapView props. 

### PR DESCRIPTION
**Pull request by dorthwein**

### CODE REVIEW

File: example/src/examples/EarthQuakes.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ class EarthQuakes extends React.Component {
 
             <MapboxGL.CircleLayer
               id="singlePoint"
-              filter={['!' ['has', 'point_count']]}
+              filter={['!'[('has', 'point_count')]]}
```
Comment (mfazekas) [inline example/src/examples/EarthQuakes.js]: I think a comma is missing here and not sure about parenthesis. This also seems to be unrelated to the PR, was this made by ESLint?

File: javascript/components/MapView.js [hunk 2]
```diff
@@ -70,6 +70,11 @@ class MapView extends NativeBridgeComponent {
      */
     styleURL: PropTypes.string,
 
+    /**
+     * Preferred Frame Rate per second for native map renders.
```
Comment (mfazekas) [inline javascript/components/MapView.js]: Can we add more descriptive documentation?

Like the one from native (what part aplies):

https://docs.mapbox.com/ios/api/maps/5.1.1/Classes/MGLMapView.html#/c:objc(cs)MGLMapView(py)preferredFramesPerSecond

https://docs.mapbox.com/android/api/map-sdk/7.0.1/com/mapbox/mapboxsdk/maps/MapView.html#setMaximumFps-int-

File: javascript/components/MapView.js [hunk 3]
```diff
@@ -224,6 +229,7 @@ class MapView extends NativeBridgeComponent {
     attributionEnabled: true,
     logoEnabled: true,
     styleURL: MapboxGL.StyleURL.Street,
+    preferredFramesPerSecond: 60,
```
Comment (mfazekas) [inline javascript/components/MapView.js]: I don't like the default 60 fps value. According to the docs the current default is MGLMapViewPreferredFramesPerSecondDefault which is either 30 or 60 fps depending on device, can we keep that feature?

**Comment by mfazekas** [comment]
@dorthwein can you add an usecase to the PR description?

File: javascript/components/MapView.js (hunk 3 above)
Comment (dorthwein) [reply javascript/components/MapView.js]: My understanding of the docs is that on iOS the preferredRate is what it will aim for, not necessarily be at.  In the case on android, the internals are actually setting the Max Frame Rate.  So if the device doesn't support 60, it wouldn't be able to achieve 60 and would utilize the default.  I believe these settings are both controlled by the MapBox Framework.

Comment (dorthwein) [reply javascript/components/MapView.js]: Let me know if your understanding is different, Happy to change it to a different value also.

File: example/src/examples/EarthQuakes.js (hunk 1 above)
Comment (dorthwein) [reply example/src/examples/EarthQuakes.js]: def an eslint situation.  I can revert it back.

File: javascript/components/MapView.js (hunk 2 above)
Comment (dorthwein) [reply javascript/components/MapView.js]: sure

File: javascript/components/MapView.js (hunk 3 above)
Comment (dorthwein) [reply javascript/components/MapView.js]: I removed the default value and double checked on iOS & Android that it would resolve to the defaults (which it does).

File: example/src/examples/EarthQuakes.js (hunk 1 above)
Comment (dorthwein) [reply example/src/examples/EarthQuakes.js]: fixed

File: javascript/components/MapView.js (hunk 2 above)
Comment (dorthwein) [reply javascript/components/MapView.js]: updated docs

**Comment by dorthwein** [comment]
@mfazeka we currently have a turn by turn navigation app.  Without this functionality, when the map is in follow mode, it runs the processor at near 100%.  Being able to adjust this has also helped with not over taxing lower quality Android phones.  

This has been an issue for sometime.  More details can be found [here](https://github.com/mapbox/mapbox-gl-native/issues/14588)
